typescript/cli/src/commands/registry.ts

```diff
@@ -88,3 +92,88 @@ const addressesCommand: CommandModuleWithContext<{ name: string }> = {
     }
   },
 };
+
+const chainTargetsCommandOption: Options = {
+  type: 'string',
+  description: 'Comma-separated list of chain names',
+  alias: 'c',
+  // TODO choices: ... can we configure this so that it's a list of chain names including any new chains added to the local registry
+};
+
+const environmentCommandOption: Options = {
+  type: 'string',
+  description: 'The name of the environment to deploy to',
+  alias: 'e',
+  choices: ChainTypes,
+};
+
+const createAgentConfigCommand: CommandModuleWithContext<{
+  chains?: string;
+  environment?: string;
+  out: string;
+}> = {
+  command: 'agent-config',
+  describe: 'Create a new agent config',
+  builder: {
+    chains: chainTargetsCommandOption,
+    environment: environmentCommandOption,
+    out: outputFileCommandOption(
+      './configs/agent-config.json',
+      false,
+      'The path to output an agent config JSON file.',
+    ),
+  },
+  // TODO: make chains and environment mutually exclusive, but require one of them
+  // builder: (yargs: Argv<{}>) =>
+  //   yargs
+  //     .option('chains', chainTargetsCommandOption)
+  //     .option('environment', environmentCommandOption)
+  //     .option(
+  //       'config',
+  //       outputFileCommandOption(
+  //         './configs/agent-config.json',
+  //         false,
+  //         'The path to output an agent config JSON file.',
+  //       ),
+  //     )
+  //     .check((argv) => {
+  //       if (!argv.chains && !argv.environment) {
+  //         throw new Error(
+  //           'Either --chains or --environment must be specified.',
+  //         );
+  //       }
+  //       return true;
+  //     })
+  //     .check((argv) => {
+  //       if (argv.chains && argv.environment) {
+  //         throw new Error(
+  //           '--chains and --environment cannot be specified together.',
+  //         );
+  //       }
+  //       return true;
+  //     }),
+  handler: async ({ context, chains, environment, out }) => {
+    const { registry } = context;
+
+    let chainNames;
+    if (chains) {
+      chainNames = chains.split(',');
+      const validChainNames = await registry.getChains();
+
+      const invalidChainNames = chainNames.filter(
+        (chainName) => !validChainNames.includes(chainName),
+      );
```

Comment:
I think `multiProvider.hasChain` is what you're looking for here.
You can get the MP from context